### PR TITLE
Enable / Disable apparmor feature using deploy tool

### PR DESCRIPTION
**Pull request by MahalakshmiManthiram**
Enable / Disable apparmor feature using deploy tool

* Added Apparmor attribute to support the security in deployment manager

This commit add support for enabling and disabling appArmor in deployment manager

Test Plan:

-  make && DEBUG=yes ran successfully
-  Tested in Simplex system by adding appArmor field in config file and applied it and verified using system host-show command
-  Tested in simplex system by removing the config file to let the deploy tool  generate new config file which gets apparmor value from system configuration

Signed-off-by: mmanthir <Mahalakshmi.Manthiram@windriver.com>

### CODE REVIEW

**Review by yjian118 — CHANGES_REQUESTED** [review]
This is a change in the of the CRDs, need to make helm chart and include the change of the chart, index etc in this commit

File: helm/wind-river-cloud-platform-deployment-manager/templates/crds.yaml [hunk 1]
```diff
@@ -578,6 +578,9 @@ spec:
                 description: MaxCPUMhzConfigured defines the maximum limit of the CPU mhz configured on the host.
                 pattern: ^[1-9][0-9]*$
                 type: string
+              appArmor:
+                description: AppArmor defines the security model of the host
+                type: string
```
Comment (yjian118) [inline helm/wind-river-cloud-platform-deployment-manager/templates/crds.yaml:583]: I suggest the type be boolean. As this this CR is configured as "enabled/disabled"

File: api/v1/hostprofile_types.go [hunk 2]
```diff
@@ -850,6 +850,10 @@ type ProfileBaseAttributes struct {
 	// +kubebuilder:validation:Pattern=^[1-9][0-9]*$
 	// +optional
 	MaxCPUMhzConfigured *string `json:"maxCPUMhzConfigured,omitempty"`
+
+	// AppArmor defines the security model on the host.
+	// +optional
+	AppArmor *string `json:"appArmor,omitempty"`
```
Comment (yjian118) [inline api/v1/hostprofile_types.go:856]: How about set it as boolean

Comment (MahalakshmiManthiram) [reply api/v1/hostprofile_types.go:856]: System CLI / REST API have apparmor as string. So leaving it as string

File: helm/wind-river-cloud-platform-deployment-manager/templates/crds.yaml (hunk 1 above)
Comment (MahalakshmiManthiram) [reply helm/wind-river-cloud-platform-deployment-manager/templates/crds.yaml]: System CLI / REST API have apparmor as string. So leaving it as string

File: go.mod [hunk 3]
```diff
@@ -7,6 +7,7 @@ require (
 	github.com/fsnotify/fsnotify v1.5.1
 	github.com/ghodss/yaml v1.0.0
 	github.com/go-logr/logr v1.2.0
+	github.com/gophercloud/gophercloud v0.0.0-20221230131953-acb6bd9c1992
```
Comment (yjian118) [inline go.mod:10]: The gophercloud reference doesn't match the reference in the replace section

File: go.mod [hunk 4]
```diff
@@ -25,7 +26,7 @@ require (
 	cloud.google.com/go v0.81.0 // indirect
 	github.com/Azure/go-autorest v14.2.0+incompatible // indirect
 	github.com/Azure/go-autorest/autorest v0.11.18 // indirect
-	github.com/Azure/go-autorest/autorest/adal v0.9.13 // indirect
+	github.com/Azure/go-autorest/autorest/adal v0.9.22 // indirect
```
Comment (yjian118) [inline go.mod:29]: We normally don't update the vendor package only if there's a specific reason

File: go.mod [hunk 5]
```diff
@@ -36,13 +37,13 @@ require (
 	github.com/form3tech-oss/jwt-go v3.2.3+incompatible // indirect
 	github.com/go-logr/zapr v1.2.0 // indirect
 	github.com/gogo/protobuf v1.3.2 // indirect
+	github.com/golang-jwt/jwt/v4 v4.0.0 // indirect
```
Comment (yjian118) [inline go.mod:40]: We normally don't update the vendor package only if there's a specific reason

File: go.mod [hunk 6]
```diff
@@ -65,7 +66,7 @@ require (
 	go.uber.org/atomic v1.7.0 // indirect
 	go.uber.org/multierr v1.6.0 // indirect
 	go.uber.org/zap v1.19.1 // indirect
-	golang.org/x/crypto v0.0.0-20210817164053-32db794688a5 // indirect
+	golang.org/x/crypto v0.0.0-20220722155217-630584e8d5aa // indirect
```
Comment (yjian118) [inline go.mod:69]: We normally don't update the vendor package only if there's a specific reason

File: go.sum [hunk 7]
```diff
@@ -46,6 +46,8 @@ github.com/Azure/go-autorest/autorest v0.11.18 h1:90Y4srNYrwOtAgVo3ndrQkTYn6kf1E
 github.com/Azure/go-autorest/autorest v0.11.18/go.mod h1:dSiJPy22c3u0OtOKDNttNgqpNFY/GeWa7GH/Pz56QRA=
 github.com/Azure/go-autorest/autorest/adal v0.9.13 h1:Mp5hbtOePIzM8pJVRa3YLrWWmZtoxRXqUEzCfJt3+/Q=
 github.com/Azure/go-autorest/autorest/adal v0.9.13/go.mod h1:W/MM4U6nLxnIskrw4UwWzlHfGjwUS50aOsc/I3yuU8M=
+github.com/Azure/go-autorest/autorest/adal v0.9.22 h1:/GblQdIudfEM3AWWZ0mrYJQSd7JS4S/Mbzh6F0ov0Xc=
```
Comment (yjian118) [inline go.sum:49]: We normally don't update the vendor package only if there's a specific reason

File: go.sum [hunk 8]
```diff
@@ -46,6 +46,8 @@ github.com/Azure/go-autorest/autorest v0.11.18 h1:90Y4srNYrwOtAgVo3ndrQkTYn6kf1E
...
+github.com/Azure/go-autorest/autorest/adal v0.9.22/go.mod h1:XuAbAEUv2Tta//+voMI038TrJBqjKam0me7qR+L8Cmk=
```
Comment (yjian118) [inline go.sum:50]: We normally don't update the vendor package only if there's a specific reason

File: go.sum [hunk 9]
```diff
@@ -175,6 +185,8 @@ github.com/gogo/protobuf v1.2.1/go.mod h1:hp+jE20tsWTFYpLwKvXlhS1hjn+gTNwPg2I6zV
 github.com/gogo/protobuf v1.3.1/go.mod h1:SlYgWuQ5SjCEi6WLHjHCa1yvBfUnHcTbrrZtXPKa29o=
 github.com/gogo/protobuf v1.3.2 h1:Ov1cvc58UF3b5XjBnZv7+opcTcQFZebYjWzi34vdm4Q=
 github.com/gogo/protobuf v1.3.2/go.mod h1:P1XiOD3dCwIKUDQYPy72D8LYyHL2YPYrpS2s69NZV8Q=
+github.com/golang-jwt/jwt/v4 v4.0.0 h1:RAqyYixv1p7uEnocuy8P1nru5wprCh/MH2BIlW5z5/o=
```
Comment (yjian118) [inline go.sum:188]: Don't update the vendor pages w/o a reason

File: go.sum [hunk 10]
```diff
@@ -532,6 +544,8 @@ golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9/go.mod h1:LzIPMQfyMNhhGPh
 golang.org/x/crypto v0.0.0-20201002170205-7f63de1d35b0/go.mod h1:LzIPMQfyMNhhGPhUkYOs5KpL4U8rLKemX1yGLhDgUto=
 golang.org/x/crypto v0.0.0-20210817164053-32db794688a5 h1:HWj/xjIHfjYU5nVXpTM0s39J9CbLn7Cc5a7IC5rwsMQ=
 golang.org/x/crypto v0.0.0-20210817164053-32db794688a5/go.mod h1:GvvjBRRGRdwPK5ydBHafDWAxML/pGHZbMvKqRZ5+Abc=
+golang.org/x/crypto v0.0.0-20220722155217-630584e8d5aa h1:zuSxTR4o9y82ebqCUJYNGJbGPo6sKVl54f/TVDObg1c=
```
Comment (yjian118) [inline go.sum:547]: Don't update the vendor pages w/o a reason

File: docs/charts/index.yaml [hunk 11]
```diff
@@ -3,12 +3,12 @@ entries:
...
     description: A Helm chart for the Deployment Manager
-    digest: aa1b91cf45a83f378b52c3885a0828f4f7977c81b637b0182f3bece1359d1d1a
+    digest: 0d2a63de46fe9e6847b968e9f77b8a9615d224c936f4e2bfb48b83451c925fde
     icon: https://www.windriver.com/resources/images/wr-logo-red-2018.png
     name: wind-river-cloud-platform-deployment-manager
     urls:
     - wind-river-cloud-platform-deployment-manager-2.0.10.tgz
     version: 2.0.10
-generated: "2022-12-15T12:33:43.676153918-05:00"
+generated: "2023-01-13T10:23:42.58872577-05:00"
```
Comment (yjian118) [inline docs/charts/index.yaml:14]: The gophercloud package is updated in 20230119, but the chart is updated in 20230113, which is not preferred, can you revert your change in the chart and index, and issue a make to re-generate the chart?

**Comment by yjian118** [comment]
Need some clean-up in the go.mod and go.sum. Can you edit in the title of the PR, to give the info regarding your test steps/ system type/ system configuration?
Need a re-test after cleaning up the package reference and helm chart.
Need to offer an example configuration in the examples/ directory.

File: docs/charts/index.yaml (hunk 11 above)
Comment (MahalakshmiManthiram) [reply docs/charts/index.yaml:14]: Regenerated the helm chart. Updated the chart and index

File: go.mod (hunk 4 above)
Comment (MahalakshmiManthiram) [reply go.mod:29]: Removed the update for the vendor package

File: go.mod (hunk 5 above)
Comment (MahalakshmiManthiram) [reply go.mod:40]: Removed the update for the vendor package

File: go.mod (hunk 6 above)
Comment (MahalakshmiManthiram) [reply go.mod:69]: Removed the update for the vendor package

File: go.sum (hunk 7 above)
Comment (MahalakshmiManthiram) [reply go.sum:49]: Removed the update for the vendor package

File: go.sum (hunk 8 above)
Comment (MahalakshmiManthiram) [reply go.sum:50]: Removed the update for the vendor package

File: go.sum [hunk 12]
```diff
@@ -56,11 +58,19 @@ github.com/Azure/go-autorest/tracing v0.6.0 h1:TYi4+3m5t6K48TGI9AUdb+IzbnSxvnvUM
 github.com/Azure/go-autorest/tracing v0.6.0/go.mod h1:+vhtPC754Xsa23ID7GlGsrdKBpUA79WCAKPPZVC2DeU=
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
 github.com/BurntSushi/xgb v0.0.0-20160522181843-27f122750802/go.mod h1:IVnqGOEym/WlBOVXweHU+Q+/VP0lqqI8lqeDx9IjBqo=
+github.com/MahalakshmiManthiram/gophercloud v0.0.0-20221230131953-acb6bd9c1992 h1:DFTrCs+tYG5edXliDMJPuZkalCT/CqA+P6fYGZwgYbc=
```
[the remark this answers is not in the document]
Comment (MahalakshmiManthiram) [reply go.sum:61]: Removed all of my fork references

File: go.sum (hunk 9 above)
Comment (MahalakshmiManthiram) [reply go.sum:188]: Removed the update for the vendor package

File: go.sum (hunk 10 above)
Comment (MahalakshmiManthiram) [reply go.sum:547]: Removed the update for the vendor package

File: go.sum [hunk 13]
```diff
@@ -614,6 +628,7 @@ golang.org/x/net v0.0.0-20210428140749-89ef3d95e781/go.mod h1:OJAsFXCWl8Ukc7SiCT
 golang.org/x/net v0.0.0-20210525063256-abc453219eb5/go.mod h1:9nx3DQGgdP8bBQD5qxJ1jj9UTztislL4KSBs9R2vV5Y=
 golang.org/x/net v0.0.0-20210805182204-aaa1db679c0d/go.mod h1:9nx3DQGgdP8bBQD5qxJ1jj9UTztislL4KSBs9R2vV5Y=
 golang.org/x/net v0.0.0-20210825183410-e898025ed96a/go.mod h1:9nx3DQGgdP8bBQD5qxJ1jj9UTztislL4KSBs9R2vV5Y=
+golang.org/x/net v0.0.0-20211112202133-69e39bad7dc2/go.mod h1:9nx3DQGgdP8bBQD5qxJ1jj9UTztislL4KSBs9R2vV5Y=
```
[the remark this answers is not in the document]
Comment (MahalakshmiManthiram) [reply go.sum:631]: Removed the update for the vendor package

File: examples/aio-sx/app-armor/app-armor.yaml [hunk 14]
```diff
@@ -0,0 +1,9 @@
+apiVersion: starlingx.windriver.com/v1
```
Comment (yjian118) [inline examples/aio-sx/app-armor/app-armor.yaml:1]: We normally require a patch applied to a specific type of system for example. You'll need to create a patch based on aio-sx.yaml, and when you issue a make, the patch will be applied to create a new configuration by kusomize tool

File: examples/standard/app-armor/app-armor.yaml [hunk 15]
```diff
@@ -0,0 +1,9 @@
+apiVersion: starlingx.windriver.com/v1
```
Comment (yjian118) [inline examples/standard/app-armor/app-armor.yaml:1]: Same comment as the aiosx example. Need a patch to apply a new example with the basic fields

File: examples/aio-sx/app-armor/app-armor.yaml (hunk 14 above)
Comment (MahalakshmiManthiram) [reply examples/aio-sx/app-armor/app-armor.yaml:1]: Added the required configuration

File: examples/standard/app-armor/app-armor.yaml (hunk 15 above)
Comment (MahalakshmiManthiram) [reply examples/standard/app-armor/app-armor.yaml:1]: Added the required configuration

**Review by albailey-wr — APPROVED** [review]

**Pull request merged**